### PR TITLE
Add random transmit power and increase the maximum power

### PR DESCRIPTION
**Pull request by djylb**
Add random signal transmission power to increase tracking difficulty. Increase transmit power to +21dbm.

https://docs.espressif.com/projects/esp-idf/en/stable/esp32c3/api-reference/bluetooth/controller_vhci.html#_CPPv4N17esp_power_level_t15ESP_PWR_LVL_P21E 

enumerator ESP_PWR_LVL_P21

Corresponding to +21dbm

### CODE REVIEW

**Comment by ckcr4lyf** [comment]
Thanks for the PR. I will try and test it out on my ESP32C3 in the next few days.

File: src/main.cpp [hunk 1]
```diff
@@ -7,6 +7,8 @@
 #include <BLEUtils.h>
 #include <BLEServer.h>
 
+#include <esp_gap_ble_api.h>
```
Comment (ckcr4lyf) [inline src/main.cpp:10]: Any reason for this include? I am able to compile fine without it.

**Comment by ckcr4lyf** [comment]
Good news, I am at least able to compile this. I thought it may not work since in ESP-IDF v4 `LVL9` is max, but it seems defaults are now ESP-IDF v5.

I will try and test it "in the wild" tomorrow, but not sure exactly how to compare effectiveness vs. without randomized power. Let's see...

File: src/main.cpp (hunk 1 above)
Comment (djylb) [reply src/main.cpp:10]: I introduced esp_gap_ble_api.h while experimenting with some related functionalities during testing. It seems I missed removing it during code cleanup. Now, it appears that this inclusion is no longer necessary. Thank you for pointing it out!

**Review by ckcr4lyf — APPROVED** [review]
LGTM.

**Comment by ckcr4lyf** [comment]
It does seem to be better. Thanks for the PR!

**Comment by wxh06** [comment]
`ESP_PWR_LVL_P21` is not available on ESP32 Series as its maximum power is +9dBm.

The code cannot be compiled with `board = esp32dev` or `board = m5stick-c` and hence this pull request conflicts with #33 and #36.

**Comment by djylb** [comment]
> `ESP_PWR_LVL_P21`ESP32 系列不支持此功能，因为其最大功率为 +9dBm。
> 
> 该代码无法使用`board = esp32dev`or进行编译`board = m5stick-c`，因此此拉取请求与#33和#36冲突。



> `ESP_PWR_LVL_P21`ESP32 系列不支持此功能，因为其最大功率为 +9dBm。
> 
> 该代码无法使用`board = esp32dev`或执行编译`board = m5stick-c`，因此此拉取请求与#33和#36冲突。

请使用ESP-IDF v5

**Comment by wxh06** [comment]
> 请使用ESP-IDF v5
> Please use ESP-IDF v5
>
> 参考下面连接，最大支持+21dbm
> Refer to the link below
>
> <https://docs.espressif.com/projects/esp-idf/en/stable/esp32c3/api-reference/bluetooth/controller_vhci.html#_CPPv4N17esp_power_level_t15ESP_PWR_LVL_P21E>

The documentation you are referring to is for **ESP32-C3**, which the code functions normally on.

However, the maximum [`esp_power_level_t`](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/bluetooth/controller_vhci.html#_CPPv417esp_power_level_t) in **ESP32 series** is [`ESP_PWR_LVL_P9`](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/bluetooth/controller_vhci.html#_CPPv4N17esp_power_level_t14ESP_PWR_LVL_P9E) (+9dbm).

**Comment by djylb** [comment]
> > 请使用ESP-IDF v5
> > 请使用ESP-IDF v5
> > 参考下面连接，最大支持+21dbm
> > 参考下面链接
> > https://docs.espressif.com/projects/esp-idf/en/stable/esp32c3/api-reference/bluetooth/controller_vhci.html#_CPPv4N17esp_power_level_t15ESP_PWR_LVL_P21E
> 
> 您所指的文档适用于**ESP32-C3**，代码在该文档上正常运行。
> 
> **然而， ESP32 系列**[`esp_power_level_t`](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/bluetooth/controller_vhci.html#_CPPv417esp_power_level_t)的最大值为(+9dbm)。****[`ESP_PWR_LVL_P9`](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/bluetooth/controller_vhci.html#_CPPv4N17esp_power_level_t14ESP_PWR_LVL_P9E)

已提交修复代码 #37 
Fixed code submitted #37

**Comment by ckcr4lyf** [comment]
Thanks for the report @wxh06 .

**Pull request merged**